### DISCUSSION TITLE
How do you guys use code-server exactly? How about  multiple projects?

### DISCUSSION THREAD

**Discussion by madebyjono** [coder/code-server#4125] [Q&A] [answered]
Do you spin up a single instance of it and reuse it for multiple projects? Or do you spin up different instances for different projects?

**Comment by lg-nineoneninethreenine** [comment]
As the answer for the first question in the title: code-server is VS Code in the browser (you know that) but it can be used from anywhere. Now, for example, you have a AWS EC2 Free Tier VM. You can't use VS Code easily, you need to connect via OpenSSH, get VS Code and connect it to the VM, which loads your computer up(at least for me). But code-server operates on the browser, it is lightweight, no big RAM usage. You can use `code-server --link` and sign in via GitHub to get a custom URL that you can use to access code-server from anywhere.

As answer to the second question, yes. The Workspaces feature in VS Code is also in code-server. You need to go to the Hamburger Menu button, hit it and go to the File section. From there, add a workspace with `Add Workspace` or something like that. You can also save it so that once you open it, the workspace loads up.

And, it is your choice if you want to use multiple instances. I use a single instance and if I ever get the need, use the Workspaces feature.

**Reply by madebyjono** [reply]
Thank you for taking the time to explain. I totally get how code-server works, and was mainly wondering if people run a single or multiple instances of it for different projects - this you've also explained very clearly :) Thanks again! Cheers!

**Comment by jsjoeio** [comment]
cc @bpmct

**Comment by bpmct** [comment, marked as answer]
There are advantages to both, like @im-coder-lg said. If you have all your projects on a single instance or VM, it's easy to access it from anywhere. If you have experience with Docker and containers, having a container for each project is a great way to isolate your projects and ensure your environment is consistent.

**Reply by madebyjono** [reply]
Makes sense! Having projects isolated was what I had in mind when asking the question. Will weigh the pros and cons of each and see how it goes! Thank you! 👍

**Reply by bpmct** [reply]
@madebyjono what did you end up using?

**Reply by madebyjono** [reply]
@bpmct I went with a single instance for now - definitely much easier than setting up multiple instances.  Will see how it goes as I add more projects to it! Cheers :)
